sdk/cosmos/src/authorization_policy.rs

```diff
@@ -74,7 +63,7 @@ impl Policy<CosmosContext> for AuthorizationPolicy {
             generate_authorization(
                 &self.authorization_token,
                 &request.method(),
-                &ctx.get_contents().resource_type,
+                ctx.get().expect("Cosmos pipeline bug: ResourceType must be present in the context at this point"),
```

Comment:
Using `Context` for something that's required feels wrong. It would be nice if this remained something that is checked at compile time.